hbase-server/src/test/java/org/apache/hadoop/hbase/TestClusterPortAssignment.java

```diff
@@ -67,17 +72,22 @@ public void testClusterPortAssignment() throws Exception {
         assertEquals("RS info port is incorrect", rsInfoPort,
           cluster.getRegionServer(0).getInfoServer().getPort());
       } catch (Exception e) {
-        if (e instanceof  BindException || e.getCause() != null &&
-            (e.getCause() instanceof BindException || e.getCause().getCause() != null &&
-              e.getCause().getCause() instanceof BindException)) {
+        Throwable rootCause = ExceptionUtils.getRootCause(e);
+        if (rootCause instanceof BindException) {
           LOG.info("Failed bind, need to retry", e);
           retry = true;
+          retryCount++;
         } else {
-          throw e;
+          retry = false;
+          fail("Failed to start mini cluster" + e);
```

Comment:
Rather than including Exception message with Assert.fail, let's log it separately just above this line (similar to `if` part) and here we can exclude it from `fail`, we can full stacktrace in log anyways.